### PR TITLE
[has-rock] Rock NFT voting strategy

### PR DESCRIPTION
**Pull request by meirbank**
Changes proposed in this pull request:
- Strategy to calculate scores for rock NFT holders, number 0-99.

The rock NFT contract is not compatible with ERC721 or ERC1155 and needs a unique strategy.
https://etherscan.io/address/0x37504ae0282f5f334ed29b4548646f887977b7cc

### CODE REVIEW

File: src/strategies/has-rock/index.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+       {
+          "name":"timesSold",
+          "type":"uint256"
+       }
+    ],
+    "payable":false,
+    "stateMutability":"view",
+    "type":"function"
+ }
+];
```
Comment (ChaituVR) [inline src/strategies/has-rock/index.ts:37]:
```suggestion
const abi = [
  'function rocks(uint256) view returns (address owner, bool currentlyForSale, uint256 price, uint256 timesSold)'
];
```

File: src/strategies/has-rock/index.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  const response = await multicall(
+    network,
+    provider,
+    abi,
+    calls,
+    { blockTag }
+  );
+
+  let result ={} as  any;
```
Comment (ChaituVR) [inline src/strategies/has-rock/index.ts:66]:
```suggestion
  let result = {} as  any;
```

File: src/strategies/has-rock/index.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  addresses.forEach((address, x)=> {
+    let addressRocks=0;
```
Comment (ChaituVR) [inline src/strategies/has-rock/index.ts:69]:
```suggestion
    let addressRocks = 0;
```

**Review by ChaituVR — CHANGES_REQUESTED** [review]

**Review by bonustrack — APPROVED** [review]

**Review by ChaituVR — APPROVED** [review]

**Comment by ChaituVR** [comment]
Strategy is live:  https://snapshot.org/#/strategy/has-rock

**Pull request merged**